src/sitemap/handler.js

```diff
@@ -150,12 +150,13 @@ export async function checkSitemap(sitemapUrl) {
  * @param {Array<string>} urls - Array of URLs to check.
  * @returns {Promise<Array<string>>} - List of sitemap URLs that exist.
  */
-async function checkCommonSitemapUrls(urls) {
+async function filterValidUrls(urls) {
   const fetchPromises = urls.map(async (url) => {
     const response = await fetch(url, { method: 'HEAD' });
     return response.ok ? url : null;
   });
   const results = await Promise.all(fetchPromises);
```

Comment:
A check from one promise rejected will fail all promises. We should use `.allSettled` more.
It would also be a good idea to fetch with a timeout set, like here: https://github.com/adobe/spacecat-audit-worker/blob/main/src/backlinks/handler.js#L25.
And to log the status code, so that we can introduce a Coralogix alert in case we receive status code 429, which would enable us to monitor if the crawling is checking too many URLs at once.

Reply:
Changed to allSettled and opened a separate PR https://github.com/adobe/spacecat-audit-worker/pull/357 to have fetchWithTimeout fct extensible.